src/rust/engine/src/nodes.rs

```diff
@@ -198,6 +156,21 @@ impl Select {
       return future::ok(value.clone()).to_boxed();
     }
 
+    fn select(select: &Select, context: &Context, product: TypeConstraint) -> NodeFuture<Value> {
+      let edges = context
+        .core
+        .rule_graph
+        .edges_for_inner(&select.entry)
+        .ok_or_else(|| {
+          throw(&format!(
+            "Tried to select product {} for intrinsic but found no edges",
```

Comment:
This mentions selecting for an intrinsic, but this method doesn't look intrinsic specific. Should maybe put `intrinsic` in the name of the method.

Reply:
Moved this from being an inner function to being a struct-level function, pulled out param for description